x/inter-tx/keeper/msg_server_test.go

```diff
@@ -78,3 +79,100 @@ func (suite *KeeperTestSuite) TestRegisterInterchainAccount() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestSubmitTx() {
+	var (
+		path *ibctesting.Path
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success", func() {
+			}, true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			icaAppA := suite.GetICAApp(suite.chainA)
+			icaAppB := suite.GetICAApp(suite.chainB)
+
+			owner := TestOwnerAddress // must be explicitly changed
+			addr, err := sdk.AccAddressFromBech32(owner)
+			suite.Require().NoError(err)
+
+			path = NewICAPath(suite.chainA, suite.chainB)
+			suite.coordinator.SetupConnections(path)
+
+			err = SetupICAPath(path, icaAppA, owner)
+			suite.Require().NoError(err)
+
+			portID, err := icatypes.GeneratePortID(TestOwnerAddress, ibctesting.FirstConnectionID, ibctesting.FirstConnectionID)
+			suite.Require().NoError(err)
+
+			// Get the address of the interchain account stored in state during handshake step
+			interchainAccountAddr, found := suite.GetICAApp(suite.chainA).ICAControllerKeeper.GetInterchainAccountAddress(suite.chainA.GetContext(), portID)
+			suite.Require().True(found)
+
+			icaAddr, err := sdk.AccAddressFromBech32(interchainAccountAddr)
+			suite.Require().NoError(err)
+
+			// Check if account is created
+			interchainAccount := icaAppB.AccountKeeper.GetAccount(suite.chainB.GetContext(), icaAddr)
+			suite.Require().Equal(interchainAccount.GetAddress().String(), interchainAccountAddr)
+
+			// Fund interchain account
+			suite.fundICAWallet(interchainAccountAddr, sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(10000))))
+
+			// Create bank transfer message to execute on the host
+			icaMsg := &banktypes.MsgSend{
+				FromAddress: interchainAccountAddr,
+				ToAddress:   suite.chainB.SenderAccount.GetAddress().String(),
+				Amount:      sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))),
+			}
+
+			// Get initial balance of recipient account on host chain
+			initialBalance := icaAppB.BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+
+			msgSrv := keeper.NewMsgServerImpl(icaAppA.InterTxKeeper)
+			msg, err := types.NewMsgSubmitTx(addr, icaMsg, path.EndpointA.ConnectionID, path.EndpointB.ConnectionID)
+			suite.Require().NoError(err)
+
+			res, err := msgSrv.SubmitTx(sdk.WrapSDKContext(suite.chainA.GetContext()), msg)
+			suite.Require().NoError(err)
+
+			// TODO
```

Comment:
This needs to relay the packet. I would suggest calling `suite.chainA.SendMsgs` as all the testing [handshake functions](https://github.com/cosmos/ibc-go/blob/92f8bb3f98db33eb8f15036feace58431a228a84/testing/endpoint.go#L325) do. This will commit the state changes to the blockchain. Calling the message server directly won't commit a block. The block needs to be committed for the packet to be relayed

To relay the packet, you may be able to use this [function](https://github.com/cosmos/ibc-go/blob/92f8bb3f98db33eb8f15036feace58431a228a84/testing/path.go#L41). It might be cool to add a different function which relays off the events emitted, but I'm not sure if events are easily accessible from within the go code